subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/dependencies/DefaultMinimalDependency.java

```diff
@@ -72,6 +72,9 @@ private int doComputeHashCode() {
 
     @Override
     public String toString() {
-        return module + ":" + versionConstraint;
+        String versionConstraintAsString = versionConstraint.toString();
```

Comment:
Why did you do this and the similar change in `DefaultPluginDependency`?
I am a bit concerned about potential side effects.

Reply:
I noticed that if you use `withoutVersion` in version catalog, then toString returns `group-id:artifact-id:` (note the `:` at the end), with this change it returns `group-id:artifact-id`. 

ToString method for this class and `DefaultPluginDependency` was added in https://github.com/gradle/gradle/pull/18218 so it should not break anything.